docs/Migrations.md

```diff
@@ -46,4 +46,112 @@ Migrations which are Node modules should export a function which will be called
+Once an async migration starts, it keeps running forever even after Pine instance restarts. The async migration will stop being executed when finalized with a dedicated flag in the async migration definition. This is to guarantee that new data inserted or updated at runtime can also be migrated. The status of running migrations can be checked in `migration status` table. The configured execution parameters and the execution metrics are stored and updated after each migration batch completes. 
+
+Each async migration needs to specify a pair of an async and a sync migration part, so that the sync migration statement closes the async migration and guarantees database consistency.
+
+Async migrations are stored in the migrations folder, alongside synchronous migrations. Their file names are significant and must contain the string `.async.` so they are treated as async migrations. They can only be Typescript / Javascript (with `.ts`/`.js` extensions) files.
+
+The async migration query must have a `LIMIT` statement to limit the maximum number of affected rows per batch.
+
+
+### Async migration procedure
+An async migration procedure for adding / renaming a column may follow these steps:
+1. Deployment
+    - Add new column (with independent sync migration) to contain new data and add code accessing the new column
+	- Old code reads/writes data to/from old column and async migration migrates data from old column to new column
+	- New code reads/writes to/from new column
+	- Async migrator runs forever
+2. Deployment
+	- Finalize async migration => only sync migration part gets executed
+	- Sync migration migrates all left over data from old column to new column
+	- Sync migration deletes old column
+	- Only new code with reads/writes to/from new column 
+	- Disable all old code that reads/writes to old column
+
+### TS migration file format with SQL query string
+
+The placeholder `%%ASYNC_BATCH_SIZE%%` will be replaced with the value specified by asyncBatchSize parameter
+
+``` javascript
+export = {
+	asyncSql: `\
+        UPDATE "device"
+        SET "note" = "device"."name"
+        WHERE id IN (   SELECT id FROM (
+                                                SELECT id FROM "device"
+                                                WHERE  "device"."name" <> "device"."note" OR "device"."note" IS NULL
+                                                LIMIT %%ASYNC_BATCH_SIZE%%
+                                                ) tmp
+                                );
+        `,
+	syncSql: `\
+        UPDATE "device"
+        SET "note" = "device"."name"
+        WHERE id IN (   SELECT id FROM (
+                                                SELECT id FROM "device"
+                                                WHERE  "device"."name" <> "device"."note" OR "device"."note" IS NULL
+                                                ) tmp
+                                );
+        `,
```

Comment:
Do we really need the `tmp` subselect?
```suggestion
	asyncSql: `\
	UPDATE "device"
	SET "note" = "device"."name"
	WHERE id IN (
		SELECT id FROM "device"
		WHERE  "device"."name" <> "device"."note" OR "device"."note" IS NULL
		LIMIT %%ASYNC_BATCH_SIZE%%
	);
	`,
	syncSql: `\
	UPDATE "device"
	SET "note" = "device"."name"
	WHERE  "device"."name" <> "device"."note" OR "device"."note" IS NULL;
	`,
```
I think the indentation was weird as well. Mind the mixed spaced & tabs.